Source/Core/EllipsoidRhumbLine.js

```diff
@@ -0,0 +1,533 @@
+        //This equation will diverge if heading gets close to 90 degrees
+        if (CesiumMath.equalsEpsilon(Math.abs(heading), CesiumMath.PI_OVER_TWO, CesiumMath.EPSILON8)) { //If heading is close to 90 degrees
+            if (major === minor) {
+                distance = major * Math.cos(firstLatitude) * CesiumMath.negativePiToPi(deltaLongitude);
+            } else {
+                var sinPhi = Math.sin(firstLatitude);
+                distance = major * Math.cos(firstLatitude) * CesiumMath.negativePiToPi(deltaLongitude) / Math.sqrt(1 - ellipsoidRhumbLine._ellipticitySquared * sinPhi * sinPhi);
+            }
+        } else {
+            var M1 = calculateM(ellipsoidRhumbLine._ellipticity, major, firstLatitude);
+            var M2 = calculateM(ellipsoidRhumbLine._ellipticity, major, secondLatitude);
+
+            distance = (M2 - M1) / Math.cos(heading);
+        }
+        return Math.abs(distance);
+    }
+
+    var scratchCart1 = new Cartesian3();
+    var scratchCart2 = new Cartesian3();
+
+    function initialize(ellipsoidRhumbLine, start, end, ellipsoid) {
+        var major = ellipsoid.maximumRadius;
+        var minor = ellipsoid.minimumRadius;
+        var majorSquared = major * major;
+        var minorSquared = minor * minor;
+        ellipsoidRhumbLine._ellipticitySquared = (majorSquared - minorSquared) / majorSquared;
+        ellipsoidRhumbLine._ellipticity = Math.sqrt(ellipsoidRhumbLine._ellipticitySquared);
+
+        if (defined(start)) {
+            ellipsoidRhumbLine._start = Cartographic.clone(start, ellipsoidRhumbLine._start);
+            ellipsoidRhumbLine._start.height = 0;
+        }
+        if (defined(end)) {
+            ellipsoidRhumbLine._end = Cartographic.clone(end, ellipsoidRhumbLine._end);
+            ellipsoidRhumbLine._end.height = 0;
+        }
+    }
+
+    function computeProperties(ellipsoidRhumbLine, start, end, ellipsoid) {
+        var firstCartesian = Cartesian3.normalize(ellipsoid.cartographicToCartesian(start, scratchCart2), scratchCart1);
+        var lastCartesian = Cartesian3.normalize(ellipsoid.cartographicToCartesian(end, scratchCart2), scratchCart2);
+
+        //>>includeStart('debug', pragmas.debug);
+        Check.typeOf.number.greaterThanOrEquals('value', Math.abs(Math.abs(Cartesian3.angleBetween(firstCartesian, lastCartesian)) - Math.PI), 0.0125);
+        //>>includeEnd('debug');
+
+        initialize(ellipsoidRhumbLine, start, end, ellipsoid);
+
+        ellipsoidRhumbLine._heading = calculateHeading(ellipsoidRhumbLine, start.longitude, start.latitude, end.longitude, end.latitude);
+        ellipsoidRhumbLine._distance = calculateArcLength(ellipsoidRhumbLine, ellipsoid.maximumRadius, ellipsoid.minimumRadius,
+                                                          start.longitude, start.latitude, end.longitude, end.latitude);
+    }
+
+    /**
+     * Initializes a rhumb line on the ellipsoid connecting the two provided planetodetic points.
+     *
```

Comment:
Should document behavior when start and end are on opposite meridians (returns E-W line).    Should there be a mechanism to get the opposite line?

Reply:
@GatorScott This check does not allow points to be on opposite ends meridians (same limitation as `EllipsoidGeodesic`).

https://github.com/AnalyticalGraphicsInc/cesium/blob/c323623a52b7b7b318421ffbb09738a4d9727d5e/Source/Core/EllipsoidRhumbLine.js#L159

Reply:
Shouldn't this limitation/restriction be documented so the developer can protect themselves from this DeveloperError: "Expected value to be greater than or equal to 0.0125, actual value was 0.0"?  Secondarily, is 0.716197 degrees the right epsilon?

Reply:
I added the documentation for the developer error. I'm not entirely certain about how 0.0125 came to be, which is from the STK Components code. @fstoner anything you can add?